Framework/Algorithms/src/GetDetectorOffsets.cpp

```diff
@@ -81,12 +85,26 @@ std::map<std::string, std::string> GetDetectorOffsets::validateInputs() {
     result["InputWorkspace"] = "The InputWorkspace must be a MatrixWorkspace.";
     return result;
   }
+
   const auto unit = inputWS->getAxis(0)->unit()->caption();
-  const auto unitErrorMsg =
-      "GetDetectorOffsets only supports input workspaces with units 'Bins of Shift' or 'd-Spacing', your unit was : " +
-      unit;
   if (unit != "Bins of Shift" && unit != "d-Spacing") {
+    const auto unitErrorMsg = "GetDetectorOffsets only supports input workspaces with units 'Bins of Shift' or "
+                              "'d-Spacing', your unit was : " +
+                              unit;
     result["InputWorkspace"] = unitErrorMsg;
+    return result;
+  }
+
+  // It's somewhat "iffy" to set this here, but it cannot be set as a default value previously,
+  //   because it's dependent on the "OutputWorkspace" property value.
+  if (isDefault("MaskWorkspace"))
+    setPropertyValue("MaskWorkspace", getPropertyValue("OutputWorkspace") + "_mask");
+  MaskWorkspace_const_sptr maskWS = getProperty("MaskWorkspace");
+  if (maskWS) {
+    // detectors which are monitors are not included in the mask
+    if (!(maskWS->getNumberHistograms() == inputWS->getInstrument()->getNumberDetectors(true))) {
```

Comment:
I'm not sure this is entirely correct. It may be better to first check that the instruments match
```c++
if (maskWS->getNumberHistograms() == inputWS->getInstrument()) {
  if (maskWS->getNumberHistograms() != inputWS->getNumberHistograms()) {
    // useful error messages
  }
} else {
  result["InputWorkspace"] = "Must have same instrument as MaskWorkspace";
  result["MaskWorkspace"] = "Must have same instrument as InputWorkspace";
}